modules/registry-refactoring/src/test/java/org/apache/airavata/registry/core/repositories/appcatalog/ApplicationDeploymentRepositoryTest.java

```diff
@@ -161,23 +161,29 @@ public void ApplicationDeploymentRepositoryTest() throws AppCatalogException {
         applicationDeploymentRepository.updateApplicationDeployment(appDeploymentId , applicationDeploymentDescription);
         assertTrue(applicationDeploymentRepository.getApplicationDeployement(appDeploymentId).getDefaultQueueName().equals("queue3"));
 
-        applicationDeploymentRepository.addApplicationDeployment(applicationDeploymentDescription1, gatewayId);
+        String deploymentId1 = applicationDeploymentRepository.addApplicationDeployment(applicationDeploymentDescription1, gatewayId);
         List<ApplicationDeploymentDescription> appDeploymentList = applicationDeploymentRepository.getAllApplicationDeployements(gatewayId);
         List<String> appDeploymentIds = applicationDeploymentRepository.getAllApplicationDeployementIds();
         assertTrue(appDeploymentList.size() == 2);
         assertTrue(appDeploymentIds.size() == 2);
 
-        applicationDeploymentRepository.removeAppDeployment(applicationDeploymentDescription1.getAppDeploymentId());
-        assertFalse(applicationDeploymentRepository.isExists(applicationDeploymentDescription1.getAppDeploymentId()));
-
         Map<String, String> filters = new HashMap<>();
         filters.put(DBConstants.ApplicationDeployment.APPLICATION_MODULE_ID, applicationModule.getAppModuleId());
         filters.put(DBConstants.ApplicationDeployment.COMPUTE_HOST_ID, computeResourceDescription.getComputeResourceId());
         appDeploymentList = applicationDeploymentRepository.getApplicationDeployements(filters);
         assertEquals(computeResourceDescription.getComputeResourceId(), appDeploymentList.get(0).getComputeHostId());
```

Comment:
Is it safe to call appDeploymentList.get(0) ? You might have to check the size of the appDeploymentList first and a potential null check of appDeploymentList